nbdev/clean.py

```diff
@@ -52,29 +53,36 @@ def _clean_cell_output(cell):
             o.get('metadata', {}).pop('tags', None)
 
 # %% ../nbs/11_clean.ipynb 8
-def _clean_cell(cell, clear_all=False):
+def _clean_cell(cell, clear_all=False, allowed_metadata_keys=None):
     "Clean `cell` by removing superfluous metadata or everything except the input if `clear_all`"
     if 'execution_count' in cell: cell['execution_count'] = None
     if 'outputs' in cell:
         if clear_all: cell['outputs'] = []
         else:         _clean_cell_output(cell)
     if cell['source'] == ['']: cell['source'] = []
     cell['metadata'] = {} if clear_all else {
-        k:v for k,v in cell['metadata'].items() if k=="hide_input"}
+        k:v for k,v in cell['metadata'].items() if k in allowed_metadata_keys}
 
 # %% ../nbs/11_clean.ipynb 9
-def clean_nb(nb, clear_all=False):
+def clean_nb(nb, clear_all=False,
+             allowed_metadata_keys:List[str]=None,
```

Comment:
We don't use generic types in fastai libs - just use `list` here

Reply:
Oh also since you're putting params on separate lines, you may also want to add [docments](https://fastcore.fast.ai/docments.html) since you'll then get param docs automatically